packages/nameguard-react/src/components/ReportIcon/index.tsx

```diff
@@ -0,0 +1,95 @@
+import React from "react";
+import {
+  CheckResultCode,
+  ConsolidatedNameGuardReport,
+} from "@namehash/nameguard";
+import cc from "classcat";
+
+import { Tooltip } from "../Tooltip/Tooltip";
+import { RatingLoadingIcon } from "../icons/RatingLoadingIcon";
+import { RatingUnknownIcon } from "../icons/RatingUnknownIcon";
+import { RatingIcon, RatingIconSize } from "../Report/RatingIcon";
+import { checkResultCodeTextColor, ratingTextColor } from "../../utils/text";
+
+type ReportShieldProps = {
+  onClick?: () => any;
+  data?: ConsolidatedNameGuardReport;
+  hadLoadingError: boolean;
+  size?: RatingIconSize;
+} & React.ComponentProps;
+
+declare global {
+  interface Window {
+    location: Location;
+  }
+}
+
+export function ReportIcon({
```

Comment:
I believe each `ReportIcon` should also take a `name` param with all the same logic and rules for that as we have in `ReportBadge`. Whenever we create a `ReportIcon` we should already know what name we're creating it for.

This is important based on the rules for how `ReportIcon` should always be clickable no matter if they are loading or hadLoadingError.